Packages/MIES/MIES_Utilities.ipf

```diff
@@ -6876,3 +6876,39 @@ Function/WAVE FlattenWaves(WAVE/WAVE input, [variable sdim])
 
 	return output
 End
+
+/// @brief Return the truth if any of the given intervals ]A, B[ intersect.
+///
+/// @param rangesParam Nx2 wave with the interval borders
+threadsafe Function AreIntervalsIntersecting(WAVE rangesParam)
```

Comment:
General comment on the function:

I know it is for the ranges here but the function is generalized. So if the function name is about "intervals" the WAVE name should follow that scheme.

Also the for loop seems to be inefficient for large sets of intervals.
How about putting the check in an indexable function and call it through multithread?
The result can then be derived from the sum of the MT return wave.

Reply:
I've renamed the function parameter. And also added the MT speedup.